src/main.ts

```diff
@@ -694,6 +694,20 @@ export class Purchases {
     return this._appUserId;
   }
 
+  /**
+   * Sets attributes for the current user. Attributes are useful for storing additional, structured information on a customer that can be used elsewhere in the system.
+   * For example, you could store your customer's email address or additional system identifiers through the applicable reserved attributes, or store arbitrary facts like onboarding survey responses, feature usage, or other dimensions as custom attributes.
+   * @param attributes - A dictionary of attributes to set for the current user.
+   * @throws {@link PurchasesError} if there is an error while setting the attributes or if the customer doesn't exist.
+   */
+  public async setAttributes(attributes: {
```

Comment:
Not for this PR! After this, we might want to add smaller convenience methods for some of the reserved attributes... or if we're concerned about making multiple calls for each individual call, we can think about other options.

Reply:
I mentioned this in a comment above, couldn't we include an `enum` with reserved attribute names and do something like:

```
export enum ReservedSubscriberAttribute {
    email = "$email",
    phoneNumber = "$phoneNumber",
    displayName = "$displayName",
    pushToken = "$apnsTokens"
    
    ...
}

...

type AttributeKey = ReservedSubscriberAttribute | (string & {});

public async setAttributes(attributes: Record<AttributeKey, string | null>): Promise<void> {
}
```